spicy/toolchain/src/compiler/validator.cc

```diff
@@ -40,46 +88,15 @@ hilti::Result<hilti::Nothing> isParseableType(QualifiedType* pt, const type::uni
         if ( f->ctor() )
             return hilti::Nothing();
 
-        auto eod_attr = f->attributes()->find("&eod");
-        auto until_attr = f->attributes()->find("&until");
-        auto until_including_attr = f->attributes()->find("&until-including");
-        auto parse_at_attr = f->attributes()->find("&parse-at");
-        auto parse_from_attr = f->attributes()->find("&parse-from");
-        auto size_attr = f->attributes()->find("&size");
-        auto max_size_attr = f->attributes()->find("&max-size");
-
-        std::vector<std::string> start_attrs_present;
-        for ( const auto& i : {parse_from_attr, parse_at_attr} ) {
-            if ( i )
-                start_attrs_present.emplace_back(i->tag());
-        }
-
-        std::vector<std::string> end_attrs_present;
-        for ( const auto& i : {eod_attr, until_attr, until_including_attr} ) {
-            if ( i )
-                end_attrs_present.emplace_back(i->tag());
-        }
+        const auto& required_one_of = {"&eod", "&until", "&until-including", "&parse-at", "&parse-from", "&size"};
```

Comment:
Let's not involve lifetime extension in this, it shouldn't have any effect anyway 😅.

Also, can you leave the validated tags in the previous order to reduce churn?

```suggestion
        const auto required_one_of = {"&eod", "&parse-at", "&parse-from", "&size", "&until", "&until-including"};
```